source/common/init/target_impl.cc

```diff
@@ -73,7 +73,7 @@ TargetHandlePtr SharedTargetImpl::createHandle(absl::string_view handle_name) co
 
 bool SharedTargetImpl::ready() {
   initialized_ = true;
-  bool all_notified = true;
+  bool all_notified = !watcher_handles_.empty();
```

Comment:
Yes it is.
It seems no production code is relying on this return value. 
It's OK to leave the return value here since so far it is easy to maintain.